hnn_core/cell.py

```diff
@@ -137,37 +143,147 @@ def gid(self, gid):
+
+        # Connects sections of THIS cell together.
+        for connection in topology:
+            parent_sec = self.sections[connection[0]]
+            child_sec = self.sections[connection[2]]
+            parent_loc = connection[1]
+            child_loc = connection[3]
+            child_sec.connect(parent_sec, parent_loc, child_loc)
+
+    def build(self, p_secs, p_syn, topology, sect_loc):
+        """Build cell in Neuron.
 
-        self.soma = h.Section(cell=self, name=soma_props['name'] + '_soma')
-        self.soma.L = soma_props['L']
-        self.soma.diam = soma_props['diam']
-        self.soma.Ra = soma_props['Ra']
-        self.soma.cm = soma_props['cm']
+        Parameters
+        ----------
+        p_secs : dict
+            Dictionary with keys as section name.
+            p_secs[sec_name] is a dictionary with keys
+            L, diam, Ra, cm, syns and mech.
+            syns is a list specifying the synapses at that section.
+            The properties of syn are specified in p_syn.
+            mech is a dict with keys as the mechanism names. The
+            values are dictionaries with properties of the mechanism.
+        p_syn : dict of dict
+            Keys are name of synaptic mechanism. Each synaptic mechanism
+            has keys for parameters of the mechanism, e.g., 'e', 'tau1',
+            'tau2'.
+        topology : list of list
+            The topology of cell sections. Each element is a list of
+            4 items in the format
+            [parent_sec, parent_loc, child_sec, child_loc] where
+            parent_sec and parent_loc are float between 0 and 1
+            specifying the location in the section to connect and
+            parent_sec and child_sec are names of the connecting
+            sections.
+        sect_loc : dict of list
+            Can have keys 'proximal' or 'distal' each containing
+            names of section locations that are proximal or distal.
+
+        Examples
+        --------
+        p_secs = {
+            'soma':
+            {
+                'L': 39,
+                'diam': 20,
+                'cm': 0.85,
+                'Ra': 200.,
+                'sec_pts': [[0, 0, 0], [0, 39., 0]],
+                'syns': ['ampa', 'gabaa', 'nmda'],
+                'mechs' : {
+                    'ca': {
+                        'gbar_ca': 60
+                    }
+                }
+            }
+        }
+        """
+        if 'soma' not in p_secs:
+            raise KeyError('soma must be defined for cell')
+        self._create_sections(p_secs, topology)
+        self._create_synapses(p_secs, p_syn)
+        self._set_biophysics(p_secs)
+        self.sect_loc = sect_loc
 
     def move_to_pos(self):
         """Move cell to position."""
-        x0 = self.soma.x3d(0)
-        y0 = self.soma.y3d(0)
-        z0 = self.soma.z3d(0)
+        x0 = self.sections['soma'].x3d(0)
+        y0 = self.sections['soma'].y3d(0)
+        z0 = self.sections['soma'].z3d(0)
```

Comment:
Side note, I'm glad we're finally getting rid of `self.soma`. +1 for including it as a member of `self.sections` so that pyramidal and basket cells can each pass in all of their sections (soma and dendrites alike) in a cohesive and modular manner.